yalelibrary/blacklight-marc
# Fix fuzzy date, refine authors (repeated author), punctuation, empty date etc

- To fix fuzzy date issue: https://app.zenhub.com/workspaces/quicksearch-5e6f993ede3cd5175f6d41f2/issues/gh/yalelibrary/search-frontend/2791 ix Fuzzy date 
- To test:
-- git pull blacklight-marc PR https://github.com/yalelibrary/blacklight-marc/pull/47
-- Then go to search-frontend directory to test:
-- Go to search-frontend directory
    -- add the blacklight-marc gem to vagrant local by add this to the Vagrantfile:
          `config.vm.synced_folder "/directory of your local path where blacklight-marc saved/blacklight-marc", "/home/vagrant/blacklight-marc"`
    -- update gemfile to point the blacklight-marc to local vagrant directory:
     `gem 'blacklight-marc', path: '/home/vagrant/blacklight-marc'`
   -- bundle install
   -- vagrant up
   -- vagrant ssh search-frondend
   -- rails s -p 3000
 -- In a single search result page. Click "Cite", you will see citation styles is in the popup window.



- Fix repeated authors 
   -- example: http://localhost:3000/catalog/11200229 
    expect to see one author, compare with the prod version: https://search.library.yale.edu/catalog/11200229  repeated authors author issue  (supposed to have one author)

- Empty date with () in APA
   Example: http://localhost:3000/catalog/3315 no 
    Expect to see: no () at the beginning of APA, compare with the prod version: https://search.library.yale.edu/catalog/3315

- Fuzzy dates:
   Examples
    -- http://localhost:3000/catalog/4009139 (normal 4 digits date)
     -- http://localhost:3000/catalog/4354860 (one 4 digits date)
    -- http://localhost:3000/catalog/1234 (two dates, one copyright date, keep the one date which is not the copyright date)
    -- http://localhost:3000/catalog/201837  (1906-07)
    -- http://localhost:3000/catalog/3315 (no date in 26x C)
    -- http://localhost:3000/catalog/2644136   (189- ([1890?]
![Screen Shot 2023-02-09 at 9 51 49 AM](https://user-images.githubusercontent.com/46331329/217847830-7cf69918-dfe3-4fd6-b9f9-1cdc0fc58179.png)
    --  http://localhost:3000//catalog/2969166 (19–) ([1900?]
![Screen Shot 2023-02-09 at 9 52 04 AM](https://user-images.githubusercontent.com/46331329/217847663-95f785af-c631-46f7-879a-b2aeb6572852.png)

    -- http://localhost:3000/catalog/11437005 (between 1900 and 1950?)   [1900-1950?]
![Screen Shot 2023-02-09 at 9 52 20 AM](https://user-images.githubusercontent.com/46331329/217847715-cbaf5621-d6bb-45d2-9bec-a4734b2d1f76.png)

   -- http://localhost:3000/catalog/3085399 1881-1890  (1881-1890)

Note: The formats of file was adjusted, so there is more difference.
